File: firstrun/qwen3_coder_next/qwen3_coder_next/file_002_naive.py
from flask import Flask, send_from_directory, jsonify, request
import os
import posixpath

app = Flask(__name__)

# Configure the directory where files are stored
UPLOAD_FOLDER = 'uploads'
app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER

@app.route('/download/<path:filename>', methods=['GET'])
def download_file(filename):
    # Security check: prevent directory traversal attacks
    # Normalize the path to prevent ../ attacks
    safe_filename = posixpath.normpath(filename)
    
    # Ensure the filename doesn't contain path traversal attempts
    if '..' in safe_filename or safe_filename.startswith('/'):
        return jsonify({'error': 'Invalid filename'}), 400
    
    # Check if file exists
    filepath = os.path.join(app.config['UPLOAD_FOLDER'], safe_filename)
    
    if not os.path.exists(filepath):
        return jsonify({'error': 'File not found'}), 404
    
    try:
        # Send the file for download
        return send_from_directory(
            app.config['UPLOAD_FOLDER'],
            safe_filename,
            as_attachment=True
        )
    except Exception as e:
        return jsonify({'error': f'Error downloading file: {str(e)}'}), 500

if __name__ == '__main__':
    # Create upload directory if it doesn't exist
    os.makedirs(UPLOAD_FOLDER, exist_ok=True)
    app.run(debug=True)

File: firstrun/qwen3_coder_next/qwen3_coder_next/file_002_naive.js
const express = require('express');
const path = require('path');
const fs = require('fs');

const router = express.Router();

// Route to download a file by filename
router.get('/download/:filename', (req, res) => {
  const filename = req.params.filename;
  
  // Sanitize the filename to prevent directory traversal attacks
  const safeFilename = path.basename(filename);
  const filePath = path.join(__dirname, 'uploads', safeFilename); // Adjust path as needed
  
  // Check if file exists
  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      return res.status(404).json({ error: 'File not found' });
    }
    
    // Set headers for file download
    res.download(filePath, safeFilename, (err) => {
      if (err) {
        console.error('Download error:', err);
        res.status(500).json({ error: 'Failed to download file' });
      }
    });
  });
});

module.exports = router;